File: apps/src/chat/frontend/js/packets/http/NewContactPacket.js
import TimeUtils from "../../utils/TimeUtils.js";

export default class NewContactPacket {

    #username;
    #createdAt;

    constructor(
        username
    ){
        this.#username = username;
        this.#createdAt = TimeUtils.getCurrentTimeStamp();
    }

    toData(){
        return {
            username: this.#username,
            created_at: this.#createdAt,
            type: "new_contact"
        }
    }

    static fromData(data){
        return new NewContactPacket(
            data["username"]
        );
    }

    getUsername(){
        return this.#username;
    }

    getCreatedAt(){
        return this.#createdAt;
    }
}

File: apps/src/chat/backend/packets/http/new_contact_packet.py
from ..packet import Packet

class NewContactPacket(Packet):

    def __init__(self, username):
        self._username = username

    def get_type(self):
        return "new_contact"

    def to_data(self):
        return {
            "type": self.get_type(),
            "username": self._username
        }

    @staticmethod
    def from_data(data):
        return NewContactPacket(data["username"])

    def get_username(self) -> str:
        return self._username